include/El/core/imports/mpi.hpp

```diff
@@ -34,14 +42,79 @@ namespace mpi {
 #endif
 #endif
 
+// Yes, I realize there's some code duplication here, but it's SO MUCH
+// EASIER for the human to read I really don't care... The
+// preprocessor is the only thing that could care, but it doesn't get
+// feelings. Bwahahaha.
+
+#ifndef HYDROGEN_HAVE_ALUMINUM
+
 struct Comm
 {
     MPI_Comm comm;
-    Comm( MPI_Comm mpiComm=MPI_COMM_WORLD ) EL_NO_EXCEPT : comm(mpiComm) { }
+    Comm(MPI_Comm mpiComm=MPI_COMM_WORLD) EL_NO_EXCEPT : comm(mpiComm) { }
 
     inline int Rank() const EL_NO_RELEASE_EXCEPT;
     inline int Size() const EL_NO_RELEASE_EXCEPT;
+
 };
+
+#else
+namespace internal
+{
+struct DelayCtorType {};
+}// namespace internal
+
+struct Comm
+{
+#if defined(HYDROGEN_HAVE_AL_MPI_CUDA) || defined(HYDROGEN_HAVE_NCCL2)
+    using aluminum_comm_type = GPUBackend::comm_type;
+#else
+    using aluminum_comm_type = CPUBackend::comm_type;
+#endif // defined(HYDROGEN_HAVE_AL_MPI_CUDA) || defined(HYDROGEN_HAVE_NCCL2)
+
+    // Hack to handle global objects, MPI_COMM could be int or void*...
+    explicit Comm(internal::DelayCtorType const&,
+                  MPI_Comm mpicomm) EL_NO_EXCEPT : comm(mpicomm) {}
+#ifdef HYDROGEN_HAVE_NCCL2
+    Comm(MPI_Comm mpiComm=MPI_COMM_WORLD);
```

Comment:
Are there major performance concerns because we can't make this `EL_NO_EXCEPT`? We can probably tweak Aluminum if need be.

Reply:
I would guess they aren't *major* (creating the grid isn't exactly performance-critical). I will make the copy/move ctors explicitly `noexcept`, and that will take care of most of the performance-critical construction.